provider/src/main/kotlin/org/gradle/kotlin/dsl/resolver/SourceDistributionProvider.kt

```diff
@@ -45,6 +45,15 @@ class SourceDistributionResolver(val project: Project) : SourceDistributionProvi
     }
 
     override fun sourceDirs(): Collection<File> =
+        try {
+            collectSourceDirs()
+        } catch (e: Exception) {
+            e.printStackTrace()
```

Comment:
Is this intentional? Shouldn't it use a logger?

Reply:
Good point.

This code will run inside IntelliJ in the context of the script dependencies resolver. So `ex.printStackTrace()` will go into the void. We should report this exception up for it to be logged into the script dependencies resolver logs so we can diagnose issues properly.

I'll remove the leniency aspect from this PR so it contains the GRADLE_METADATA fix only. And follow up with another leniency PR that propagates the error up for logging.

Reply:
Actually ... you can scratch the above, I mixed up my mind.

This code will run in a Gradle daemon behind the TAPI builders that the script dependencies resolver will invoke. So it should be easy enough to use a logger to get the traces into the daemon logs.